### PR TITLE
Fix/catch exception on batch

### PR DESCRIPTION
**Pull request by joelvoss**
This PR is based on https://github.com/google-cloudsearch/norconex-committer-plugin/pull/5 and https://github.com/google-cloudsearch/norconex-committer-plugin/pull/7 as both are needed for the committer plugin to function reliably.

I've added the missing unit tests and modified the existing ones to assert logged exceptions instead of expecting a throw.

### CODE REVIEW

**Comment by donghanmiao** [comment]
Thank you Joel. We have just checked in PR#7, please rebase your commit. Thanks.

**Comment by joelvoss** [comment]
@donghanmiao , i just rebased my commits. Feel free to review. Thanks!

File: src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java [hunk 1]
```diff
@@ -104,17 +109,20 @@
...
   @Mock private XMLStreamWriter mockConfigWriter;
   @Mock private DefaultAcl mockDefaultAcl;
+  @Mock private Appender appender;
 
   @Mock(answer = Answers.RETURNS_DEEP_STUBS)
   private IndexingService mockIndexingService;
 
   @Captor private ArgumentCaptor<AbstractInputStreamContent> itemContentCaptor;
   @Captor private ArgumentCaptor<Item> itemCaptor;
+  @Captor private ArgumentCaptor<LoggingEvent> captor;
```
Comment (donghanmiao) [inline src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java]: loggingCaptor to be more specific?

File: src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java [hunk 2]
```diff
@@ -255,26 +268,30 @@ public void initShouldStartIndexingService() {
...
     setupConfig.initConfig(new java.util.Properties());
     subject.loadFromXml(mockConfig);
-    thrown.expect(CommitterException.class);
-    thrown.expectMessage(
-        "Binary content field is missing or invalid. Please configure BinaryContentTagger"
-            + " and make sure the '"
-            + GoogleCloudSearchCommitter.FIELD_BINARY_CONTENT
-            + "' field is left untouched (e.g. watch out for KeepOnlyTagger)");
     subject.commitBatch(Arrays.asList(addOperation(URL, MIME_PDF, null, !CONTENT_ERROR)));
+    verify(appender, times(8)).doAppend(captor.capture());
```
Comment (donghanmiao) [inline src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java]: verifying 'times(8)' may better be replaced with something like atLeastOnce(). It is too rigid in the current way.

File: src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java [hunk 3]
```diff
@@ -309,6 +323,11 @@ public InputStream getContentStream() throws IOException {
         throw new IOException();
       }
     }));
+    verify(appender, times(8)).doAppend(captor.capture());
```
Comment (donghanmiao) [inline src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java]: ditto.

**Review by donghanmiao — COMMENTED** [review]
Overall it looks good to me. I have a few nitpicks, please take a look. Thanks.

File: src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java (hunk 1 above)
Comment (joelvoss) [reply src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java]: makes sense. renamed to "loggingCaptor"

File: src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java (hunk 2 above)
Comment (joelvoss) [reply src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java]: agreed. changed to atLeastOnce.

File: src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java (hunk 3 above)
Comment (joelvoss) [reply src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java]: same as above. changed to atLeastOnce

File: src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java [hunk 4]
```diff
@@ -341,6 +361,21 @@ public void addShouldFailWhenContentTypeFieldIsMissing() {
     subject.commitBatch(Arrays.asList(addOperation(URL, null, CONTENT_BASE64, CONTENT_ERROR)));
   }
 
+  @Test
+  public void addShouldFailWhenBinaryContentIsEmpty() {
+    setupConfig.initConfig(new java.util.Properties());
+    subject.loadFromXml(mockConfig);
+    subject.commitBatch(Arrays.asList(addOperation(URL, MIME_PDF, null, !CONTENT_ERROR)));
+    verify(appender, times(8)).doAppend(loggingCaptor.capture());
```
Comment (donghanmiao) [inline src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java]: sorry. same here. atLeastOnce()

Comment (joelvoss) [reply src/test/java/com/norconex/committer/googlecloudsearch/GoogleCloudSearchCommitterTests.java]: my bad. fixed asap

**Pull request merged**